dare.go

```diff
@@ -44,9 +44,9 @@ const (
 
 var (
 	errMissingHeader      = errors.New("dare: incomplete header")
```

Comment:
why are we calling this `dare:` we should name it `aead:` ?  - top level callers might not make sense of this.